akka-docs/rst/java/stream/stages-overview.rst

```diff
@@ -992,12 +992,27 @@ Drop elements until a timeout has fired
 
 groupedWithin
 ^^^^^^^^^^^^^
-Chunk up the stream into groups of elements received within a time window, or limited by the given number of elements,
-whichever happens first.
+Chunk up the stream into either groups of elements limited by a given number of elements to group,
+or elements received within a time window when there's a demand from downstream.
+in case downstream hasn't pulled yet, It'll continue to chunk up elements up to the limit or until downstream pulls.
```

Comment:
But it's not entirely true. if downstream did not demand, we continue to chunk up elements.